src/styles/_catalog.scss

```diff
@@ -1,22 +1,32 @@
+$width-card: 200px;
+
 .catalog {
-  display: flex;
-  flex-wrap: wrap;
+  display: grid;
+  justify-content: center;
+  grid-template-columns: repeat(4, $width-card);
 
   max-width: 944px;
   margin: 0 auto;
+  gap: 48px;
+
+  @media (max-width: 1023px) {
+    grid-template-columns: repeat(3, $width-card);
+  }
+
+  @media (max-width: 767px) {
+    grid-template-columns: repeat(2, $width-card);
+  }
 
+  @media (max-width: 487px) {
+    grid-template-columns: repeat(1, $width-card);
+  }
```

Comment:
You can make a loop to create all media queries. Just create a correctly mapped list to iterate.